### PR TITLE
Use ResourceHttpRequestHandler in LogFileMvcEndpoint

### PR DESCRIPTION
**Pull request by joshiste**
With this change the LogFileMvcEndpoint uses a ResourceHttpRequestHandler for serving the logfile.
This gives you support for requesting parts of the logfile via the HTTP Range header.
Also requests with the If-Modified-Since header are correctly handled.

CLA is signed

### CODE REVIEW

File: spring-boot-actuator/src/test/java/org/springframework/boot/actuate/endpoint/mvc/LogFileMvcEndpointTests.java [hunk 1]
```diff
@@ -39,11 +38,15 @@
...
  * Tests for {@link LogFileMvcEndpoint}.
  *
- * @author Johannes Stelzer
+ * @author Johannes Edmeier
  * @author Phillip Webb
  */
 public class LogFileMvcEndpointTests {
 
+	/**
+	 * Comment necessary for checkstyle because this field is public - as it is with JUnit
```
Comment (wilkinsona) [inline spring-boot-actuator/src/test/java/org/springframework/boot/actuate/endpoint/mvc/LogFileMvcEndpointTests.java]: I don't think this is needed. Checkstyle's javadoc checks are disabled for everything in src/test/java

Comment (joshiste) [inline spring-boot-actuator/src/test/java/org/springframework/boot/actuate/endpoint/mvc/LogFileMvcEndpointTests.java]: This caused me an error in eclipse... I'll remove it and See if it still Compiler...

**Comment by RobWin** [comment]
@joshiste Thx for this great PR.

@philwebb The Range Header is such a nice feature. I think it would make sense to add a hint to the Spring Boot reference guide.

**Comment by wilkinsona** [comment]
@RobWin Agreed. I've opened https://github.com/spring-projects/spring-boot/issues/5062

**Comment by joshiste** [comment]
@RobWin Why don't you try a PR for the docs yourslef :grinning:

**Comment by RobWin** [comment]
Next time :)
I don't even have time enough to implement the issues of my open source projects atm. :(

**Pull request closed without merging**